app/services/api/append_metadata.rb

```diff
@@ -0,0 +1,33 @@
+# frozen_string_literal: true
+
+module Api
+  class AppendMetadata
+    def initialize(dataset)
+      @dataset = dataset
+    end
+
+    def call
+      {
+        set_name => dataset,
+        "metadata" => metadata,
```

Comment:
I think the convention here is to call this `meta`.

Reply:
Also we might want to add `links` object here with subkeys `first`, `last`, `next` and `previous`.